File: array/0075_sort_colors/0075_sort_colors.go
package main 

import "fmt"

func sortColors(nums []int) {
	i, j, k := 0, 0, len(nums) -1
	for j <= k {
		switch nums[j] {
		case 0:
			nums[i], nums[j] = nums[j], nums[i]
			i++
			j++
		case 1:
			j++
		case 2:
			nums[j], nums[k] = nums[k], nums[j]
			k--
		}
	}
}

func main() {
	nums := []int{2,0,2,1,1,0}
	sortColors(nums)
	fmt.Println(nums)
}

File: array/0075_sort_colors/0075_sort_colors.py
class Solution(object):
    def sortColors(self, nums):
        def triPartition(nums, target):
            i,j,n = 0, 0,len(nums) -1
            while j <= n:
                if nums[j] < target:
                    nums[i], nums[j] = nums[j], nums[i]
                    i += 1
                    j += 1
                elif nums[j] > target:
                    nums[j], nums[n] = nums[n], nums[j]
                    n -=1
                else:
                    j +=1

        triPartition(nums, 1)


nums = [2,0,2,1,1,0]
Solution().sortColors(nums)
print(nums)